src/UI/Seller/src/app/product-management/components/product-details-images/product-details-images.component.ts

```diff
@@ -1,12 +1,54 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Product } from '@ordercloud/angular-sdk';
+import {
+  faTrashAlt,
+  faUpload,
+  faCrown,
+} from '@fortawesome/free-solid-svg-icons';
+import { ProductUpdate } from '@app-seller/product-management/containers/product-details/product-details.component';
+import { ToastrService } from 'ngx-toastr';
 
 @Component({
   selector: 'products-details-images',
   templateUrl: './product-details-images.component.html',
   styleUrls: ['./product-details-images.component.scss'],
 })
-export class ProductDetailsImagesComponent implements OnInit {
-  constructor() {}
+export class ProductDetailsImagesComponent {
+  @Input() product: Product;
+  @Output() update = new EventEmitter<ProductUpdate>();
+  faTrash = faTrashAlt;
+  faUpload = faUpload;
+  faCrown = faCrown;
 
-  ngOnInit() {}
+  constructor(private toastrService: ToastrService) {}
+
+  deleteImage(index: number) {
+    this.product.xp.imageURLs.splice(index, 1);
+    this.update.emit(this.product);
+  }
+
+  addImage(event) {
+    const fileList: FileList = event.target.files;
+    if (fileList.length > 0) {
+      const file: File = fileList[0];
+      // Make API call to image storage integration. API should return the url at which the file is stored.
+      // Then, use commented out code below to save this URL in OrderCloud. Delete the toastr.
+
+      // const url = 'http://example.com';
+      // this.product.xp.imageURLs.push(url)
```

Comment:
will the product.xp.imageURLs always have an array at this point?, maybe some logic to check to see if an array exists first, before pushing it.